daft/context.py

```diff
@@ -161,21 +165,32 @@ def daft_planning_config(self) -> PyDaftPlanningConfig:
             return self._daft_planning_config
 
     @property
-    def runner_config(self) -> _RunnerConfig:
+    def runner_config(self) -> _RunnerConfig | None:
+        """Retrieve the currently active runner config, or None if it has not yet been set"""
         with self._lock:
-            return self._get_runner_config()
+            return self._runner_config
 
-    def _get_runner_config(self) -> _RunnerConfig:
+    def _get_or_create_runner_config(self) -> _RunnerConfig:
```

Comment:
Do you think it makes sense to initialize runner config from env upon context creation instead of it being lazy? 

This way we don't have to always check if `runner_config is None`. The planning and execution configs are also not lazy.